src/Apps/Order2/Routes/Checkout/Components/PaymentStep/Order2PaymentForm.tsx

```diff
@@ -167,7 +175,26 @@ const PaymentFormContent = ({ setConfirmationToken }) => {
       return
     }
 
-    setConfirmationToken({ confirmationToken: response.me?.confirmationToken })
+    const updateOrderPaymentMethodResult =
+      await updateOrderMutation.submitMutation({
+        variables: {
+          input: {
+            id: order.internalID,
+            paymentMethod: "CREDIT_CARD",
+          },
+        },
+      })
+
+    validateAndExtractOrderResponse(
+      updateOrderPaymentMethodResult.updateOrder?.orderOrError,
+    )
+
+    setConfirmationToken({
+      confirmationToken: {
+        id: confirmationToken.id,
+        ...response?.me?.confirmationToken,
+      },
+    })
```

Comment:
I see that this [comes from the checkout context](https://github.com/artsy/force/blob/174abc37a454d69d094ce7344cb83db946959b58/src/Apps/Order2/Routes/Checkout/Components/PaymentStep/Order2PaymentStep.tsx#L21C50-L22). In that case we don't need to pass it in as a prop.